src/main/java/com/hillel/items_exchange/service/ProductService.java

```diff
@@ -0,0 +1,18 @@
+package com.hillel.items_exchange.service;
+
+import com.hillel.items_exchange.dao.ProductRepository;
+import com.hillel.items_exchange.model.Product;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Component
+public class ProductService {
+    private final ProductRepository productRepository;
+
+    Optional<Product> findById(Long id) {
```

Comment:
What is the best practice: "unpacking" Optional on service layer or later in controller?

Reply:
It depends of your purposes. Mostly Optional returns to the place where it is needed, so I prefer to send it to Controllers layer.